cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -241,3 +243,61 @@ const char *sc_umount_cmd(char *buf, size_t buf_size, const char *target,
 
 	return buf;
 }
+
+void sc_do_mount(const char *source, const char *target,
+		 const char *fs_type, unsigned long mountflags,
+		 const void *data)
+{
+	char buf[10000];
+	const char *mount_cmd = NULL;
+
+	void ensure_mount_cmd() {
+		if (mount_cmd == NULL) {
+			mount_cmd = sc_mount_cmd(buf, sizeof buf, source,
+						 target, fs_type, mountflags,
+						 data);
+		}
+	}
+
+	if (sc_is_debug_enabled()) {
```

Comment:
Note that sc_is_debug_enabled() is still only looking at SNAP_CONFINE_DEBUG, so an attacker can set that and have the whole string handling attack surface. I thought there were plans to make this compile time?